specs/core/0_beacon-chain.md

```diff
@@ -805,44 +808,30 @@ def compute_committee(validator_indices: List[ValidatorIndex],
 
 Note: this definition and the next few definitions are highly inefficient as algorithms, as they re-calculate many sub-expressions. Production implementations are expected to appropriately use caching/memoization to avoid redoing work.
 
-### `get_crosslink_committees_at_slot`
+### `get_epoch_start_shard`
 
 ```python
-def get_crosslink_committees_at_slot(state: BeaconState,
-                                     slot: Slot) -> List[Tuple[List[ValidatorIndex], Shard]]:
-    """
-    Return the list of ``(committee, shard)`` tuples for the ``slot``.
-    """
-    epoch = slot_to_epoch(slot)
-    current_epoch = get_current_epoch(state)
-    previous_epoch = get_previous_epoch(state)
-    next_epoch = current_epoch + 1
-
-    assert previous_epoch <= epoch <= next_epoch
-    indices = get_active_validator_indices(state, epoch)
-
-    if epoch == current_epoch:
-        start_shard = state.latest_start_shard
-    elif epoch == previous_epoch:
-        previous_shard_delta = get_shard_delta(state, previous_epoch)
-        start_shard = (state.latest_start_shard - previous_shard_delta) % SHARD_COUNT
-    elif epoch == next_epoch:
-        current_shard_delta = get_shard_delta(state, current_epoch)
-        start_shard = (state.latest_start_shard + current_shard_delta) % SHARD_COUNT
+def get_epoch_start_shard(state: BeaconState, epoch: Epoch) -> Shard:
+    if epoch == get_current_epoch(state):
+        return state.latest_start_shard
+    elif epoch == get_previous_epoch(state):
+        previous_shard_delta = get_shard_delta(state, epoch)
+        return (state.latest_start_shard - previous_shard_delta) % SHARD_COUNT
+    elif epoch == get_current_epoch(state) + 1:
+        current_shard_delta = get_shard_delta(state, get_current_epoch(state))
+        return (state.latest_start_shard + current_shard_delta) % SHARD_COUNT
+    else:
+        raise Exception("Not supported")
+```
```

Comment:
```suggestion
def get_epoch_start_shard(state: BeaconState, epoch: Epoch) -> Shard:
    assert epoch <= get_current_epoch(state) + 1
    check_epoch = get_current_epoch(state) + 1
    shard = (state.latest_start_shard + get_shard_delta(state, get_current_epoch(state))) % SHARD_COUNT
    while check_epoch > epoch:
        check_epoch -= 1
        shard = (shard - get_shard_delta(state, check_epoch)) % SHARD_COUNT
    return shard
```

Reply:
Might make this `(shard + SHARD_COUNT - get_shard_delta(state, check_epoch))` just to be true to the unsigned type